### PR TITLE
Add testing to controllers

### PR DESCRIPTION
**Pull request by Josphat205**
## In this pull request

- Added RSpec test to posts and users controllers

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    before { get user_posts_path(@user) }
+
+    it('status should be :ok') do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it('should render :index template') do
+      expect(response).to render_template(:index)
+    end
+  end
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [ ] For this action please include a test for `If the response body includes correct placeholder text`.

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  context ':show' do
+    before { get user_posts_path(@user, @post) }
+
+    it('status should be :ok') do
+      expect(response).to have_http_status(:ok)
+    end
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [ ] For this section please include the tests for `If a correct template was rendered` and `If the response body includes correct placeholder text`.

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  describe ':index' do
+    before(:each) { get root_path }
+
+    it('status should be :ok') do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it('rendered :index template') do
+      expect(response).to render_template(:index)
+    end
```
Comment (thecodechaser) [inline spec/requests/users_spec.rb:13]: - [ ] For this action please include a test for `If the response body includes correct placeholder text`.

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    it('should render :show template') do
+      expect(response).to render_template(:show)
+    end
+
+    it('should assign User.find(params[:id]) to @user') do
+      expect(assigns(:user)).to eq(@author)
+    end
+  end
```
Comment (thecodechaser) [inline spec/requests/users_spec.rb]: - [ ] For this action please include a test for `If the response body includes correct placeholder text`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Josphat205👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Project Approved 👍 

Hi @Josphat205, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**